cocos/platform/CCFileUtils.h

```diff
@@ -100,6 +171,49 @@ class CC_DLL FileUtils
     virtual Data getDataFromFile(const std::string& filename);
 
     /**
+     *  Gets whole file contents as string from a file.
+     *
+     *  Unlike getStringFromFile, these methods does not truncate the string when '\0' is found.
+     *  That is thie returned string of getContents may have '\0' in it.
+     *
+     *  The template version of can accept cocos2d::Data, std::basic_string and std::vector.
+     *  For other type of buffer, please write an adapter for it, and use the non-template version.
+     *
+     *  <pre>
+     *  {@code
+     *  std::string sbuf;
+     *  FileUtils::getInstance()->getContents("path/to/file", &sbuf);
+     *
+     *  std::vector<int> vbuf;
+     *  FileUtils::getInstance()->getContents("path/to/file", &vbuf);
+     *
+     *  Data dbuf;
+     *  FileUtils::getInstance()->getContents("path/to/file", &dbuf);
+     *  }
+     * </pre
+     *
+     *  @param[in]  filename The resource file name which contains the path.
+     *  @param[out] buffer The buffer where the file contents are store to.
+     *  @return Returns:
+     *      - FileError::OK when there is no error, the buffer is filled with the contents of file.
+     *      - FileError::NotExists when file not exists, the buffer will not changed.
+     *      - FileError::OpenFailed when cannot open file, the buffer will not changed.
+     *      - FileError::ReadFaild when read end up before read whole, the buffer will fill with already read bytes.
+     *      - FileError::NotInitialized when FileUtils is not initializes, the buffer will not changed.
+     */
+    template <
+        typename T,
+        typename std::enable_if<
+            std::is_base_of< ResizableBuffer, ResizableBufferAdapter<T> >::value
+        >::type* = nullptr
```

Comment:
#5 section of http://en.cppreference.com/w/cpp/types/enable_if uses `enable_if_t` (notice the tailling `_t` ) is a C++14 feature.